pkg/kubelet/rkt/rkt.go

```diff
@@ -863,11 +898,34 @@ func (r *Runtime) generateRunCommand(pod *api.Pod, uuid string) (string, error)
 			return "", err
 		}
 	}
+
 	runPrepared = append(runPrepared, fmt.Sprintf("--hostname=%s", hostname))
 	runPrepared = append(runPrepared, uuid)
 	return strings.Join(runPrepared, " "), nil
 }
 
+func (r *Runtime) cleanupPodNetwork(pod *kubecontainer.Pod, teardown bool) error {
+	// Parse first container ID for the pod UUID
+	podID, err := parseContainerID(pod.Containers[0].ID)
+	if err != nil {
+		return fmt.Errorf("rkt: Failed to get rkt uuid of the pod %q: %v", pod.Name, err)
+	}
+
+	if teardown {
+		glog.V(3).Infof("Calling network plugin %s to tear down pod for %s", r.networkPlugin.Name(), kubecontainer.BuildPodFullName(pod.Name, pod.Namespace))
+		pluginPodID := buildContainerID(&containerID{podID.uuid, pod.Name})
+		if err := r.networkPlugin.TearDownPod(pod.Namespace, pod.Name, pluginPodID); err != nil {
```

Comment:
I had this stuff (add & plugin) before `prepare` because at prepare time environment variables are finalized, and one goal I have for this is supporting the downward api (including environment variable of the podIP). The idea to me was if you setup the namespace and run the plugins prior to preparing, you can inject the IP into prepare for `/etc/hosts` file generation and downwards api.